spec/features/idv/end_to_end_idv_spec.rb

```diff
@@ -0,0 +1,216 @@
+require 'rails_helper'
+
+RSpec.describe 'Identity verification', :js do
+  include IdvStepHelper
+
+  # Needed specs:
+  # unsupervised proofing happy path mobile only
+  # hybrid mobile end to end (edit hybrid_mobile_spec)
+  # verify by mail
+  # in person proofing
+
+  scenario 'Unsupervised proofing happy path desktop' do
+    visit_idp_from_sp_with_ial2(:oidc)
+    user = sign_up_and_2fa_ial1_user
+
+    validate_welcome_page
+    try_to_skip_ahead_from_welcome
+    complete_welcome_step
+
+    validate_agreement_page
+    try_to_skip_ahead_from_agreement
+    complete_agreement_step
+
+    validate_hybrid_handoff_page
+    try_to_skip_ahead_from_hybrid_handoff
+    complete_hybrid_handoff_step # upload photos
+
+    try_to_go_back_from_document_capture
+    validate_document_capture_page
+    complete_document_capture_step
+    validate_document_capture_submit(user)
+
+    validate_ssn_page
+    complete_ssn_step
+
+    try_to_go_back_from_verify_info
+    validate_verify_info_page
+    complete_verify_step
+    validate_verify_info_submit(user)
+
+    validate_phone_page
+    try_to_skip_ahead_from_phone
+    complete_otp_verification_page(user)
+  end
+
+  def validate_welcome_page
+    expect(page).to have_current_path(idv_welcome_path)
+
+    # Check for expected content
+    expect_step_indicator_current_step(t('step_indicator.flows.idv.getting_started'))
+  end
+
+  def validate_agreement_page
+    expect(page).to have_current_path(idv_agreement_path)
+
+    # Check for expected content
+    expect_step_indicator_current_step(t('step_indicator.flows.idv.getting_started'))
+
+    # Check for actions that shouldn't advance the user
+    # Try to continue with unchecked checkbox
+    click_continue
+    expect(page).to have_current_path(idv_agreement_path)
+    expect(page).to have_content(t('forms.validation.required_checkbox'))
+  end
+
+  def validate_hybrid_handoff_page
+    allow_any_instance_of(Idv::HybridHandoffController).to receive(:mobile_device?).
+      and_return(false)
+
+    expect(page).to have_current_path(idv_hybrid_handoff_path)
+
+    # Check for expected content
+    expect_step_indicator_current_step(t('step_indicator.flows.idv.verify_id'))
+    expect(page).to have_content(t('doc_auth.headings.upload_from_computer'))
+    expect(page).to have_content(t('doc_auth.info.upload_from_computer'))
+    expect(page).to have_content(t('doc_auth.headings.upload_from_phone'))
+
+    # defaults phone to user's 2fa phone number
+    field = page.find_field(t('two_factor_authentication.phone_label'))
+    expect(field.value).to eq('(202) 555-1212')
```

Comment:
I see it it was like this in the old style, but is there a constant or something we can pull this from? to avoid hardcoding this?

Reply:
I think that phone number comes from [here](https://github.com/18F/identity-idp/blob/ffd1c407f67e9999338e103310c98ea91481a469/spec/support/features/session_helper.rb#L39), and I can define a constant in SessionHelper and reference it in this spec, but getting the phone number formatted right is its own can of worms. Do we have spec helpers for that? I tried PhoneFormatter.format, but it gave me +1 202-555-1212 instead of (202) 555-1212.

Reply:
What if we normalize the field to be the value, so like:
```suggestion
    expect(PhoneFormatter.format(field.value)).
      to eq(PhoneFormatter.format(OUR_NEW_CONSTANT_VALUE))
```

Reply:
That works! Now where should we put a `phone_compare` method or custom matcher?

Reply:
I would do that check inline here if it's just the one? but adding a phone_compare method as a helper in this file woudl be easy to find and keep scoped